lib/aws-operations/aws-config-profile-reader.js

```diff
@@ -0,0 +1,182 @@
+/* 
+ * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+*/
+const fs = require('fs-extra')
+const chalk = require('chalk')
+const lineByLine = require('n-readlines')
+
+const awsmobileJSConstant = require('../utils/awsmobilejs-constant.js')
+const pathManager = require('../utils/awsmobilejs-path-manager.js')
+
+function getSystemConfig(profileName, silentFlag){
+    let awsConfig
+    let sysCredential = readSystemAwsCredential(profileName, silentFlag)
+    if(sysCredential){
+        awsConfig = {}
+        awsConfig.accessKeyId = sysCredential.aws_access_key_id ? sysCredential.aws_access_key_id : awsmobileJSConstant.DefaultAWSAccessKeyId
+        awsConfig.secretAccessKey = sysCredential.aws_secret_access_key ? sysCredential.aws_secret_access_key : awsmobileJSConstant.DefaultAWSSecretAccessKey
+    }
+
+    if(awsConfig){
+
+        let sysConfig = readSystemAwsConfig(profileName, silentFlag)
+
+        if(sysConfig){
+            awsConfig.region = sysConfig.region ? sysConfig.region : awsmobileJSConstant.DefaultAWSRegion
+        }else{
+            awsConfig = undefined
+        }
+    }
+
+    return awsConfig
+}
+
+function readSystemAwsConfig(profileName, silentFlag)
+{
+    let result
+
+    if(!profileName){
+        profileName = 'default'
+    }
+    let profileMark = '[' + profileName + ']'
+
+    if(profileName != 'default'){
+        profileMark = '[profile ' + profileName + ']'
+    }
+
+    let sysConfigFilePath = pathManager.getSysAwsConfigFilePath()
+
+    if(fs.existsSync(sysConfigFilePath)){
+        try{
+            let liner = new lineByLine(sysConfigFilePath)
+            let line
+            let isInProfile = false
+            while (line = liner.next()) {
+                let strPair = line.toString().split("=")
```

Comment:
Are you sure there is only one "=" on each line? Could be possible that a Key has and "=" on its value? I will try to use a library to parse that file. Could be a possible bug